python/lsst/obs/base/mapping.py

```diff
@@ -89,12 +89,15 @@ def __init__(self, datasetType, policy, registry, rootStorage, provided=None):
         self.registry = registry
         self.rootStorage = rootStorage
 
-        self.template = policy['template']  # Template path
-        self.keyDict = dict([
-            (k, _formatMap(v, k, datasetType))
-            for k, v in
-            re.findall(r'\%\((\w+)\).*?([diouxXeEfFgGcrs])', self.template)
-        ])
+        self._template = policy['template']  # Template path
+        if self._template:
+            self.keyDict = dict([
+                (k, _formatMap(v, k, datasetType))
+                for k, v in
+                re.findall(r'\%\((\w+)\).*?([diouxXeEfFgGcrs])', self.template)
+            ])
+        else:
+            self.keyDict = {}
```

Comment:
remove lines 99 and 100.

Reply:
in most cases, _template can not be used if empty, and is a property that will raise if it is gotten. In this case we do allow it to be empty for fetching the key dict so that the mapping can be constructed, so that it can raise if it's invalid. I know it's a little odd, but it allows this template check to be introduced without a major refactor. I'll add this comment to the code.